### PR TITLE
feat: adopt v7 API #WPB-11975

### PR DESCRIPTION
**Pull request by damian-kaczmarek**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11975" title="WPB-11975" target="_blank">TaskWPB-11975</a>  [Android] Adopt API v7
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

New API version added on backend https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/1309868033/API+changes+v6+v7

> [!IMPORTANT]  
> Please verify this list, as part of the CR.

Changes listed in doc (applied ✅ , not relevant ❌ )
- The capabilities field of the Client structure now contains a list of capabilities, rather than a nested object. In the notifications containing the Client structure, it will still be a nested object. ✅
- The notification contains the old format {capabilities: { capabilities: [...] }}, but clients must allow both old and new format starting from V6.  Once V5 is discontinued, the backend will start sending notifications in the new form.✅

- There is a new type of capability in capabilities field for Client: consumable-notifications. This will only show up when using API v7. In the notifications containing the Client structure, this capability will not be visible. ❌

- DELETE /oauth/applications/:id/ is removed in v7 in favor of a new endpoint DELETE /oauth/applications/:id/sessions which takes a request body with an optional password. The password is verified in case of the user is not an SSO user.❌

- renamed POST /users/handles → POST /handles❌

- renamed POST /users/handles → POST /handles❌

- renamed HEAD /users/handles → HEAD /handles❌

- renamed POST /conversations/bots → POST /bot/conversations❌

- renamed DELETE /conversations/bots → DELETE /bot/conversations❌

- renamed POST /conversations/one2one → POST /one2one-conversations ✅

- renamed GET /conversations/one2one → GET /one2one-conversations ✅

- GET /scim/auth-tokens returns an array of ScimTokenInfoitems which now have an additional field: name.❌

- POST /scim/auth-tokens response has a new optional field: name. If not specified, the name will be set to the ID of the SCIM token by the backend.❌

- GET /teams/invitations/info response has a new optional field created_by_email which will be occupied with the inviter’s email in case the invitee is a personal user.❌

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

This is more of a general play test if everything works.
There were no functional changes.

To test you need to switch to anta 
https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/300351887/Testing+environments+for+federation#anta

The html with deeplinks switching can be downloaded here. You must logout first before switching. https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/300351887/Testing+environments+for+federation#anta

To get signup code you must read the email from [inbucket](https://inbucket.anta.wire.link/monitor). 
Password is on [1password](https://start.1password.com/open/i?a=FOQLGGWZU5EQXEJO3V3A5HI3DU&v=nqjxjxdabagisaqjlyfym4kye4&i=gprcgudpieokefbatxqlxctu2m&h=wire.1password.eu).

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/E2EIApiV7.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.network.api.v7.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.api.v6.authenticated.E2EIApiV6
+
+internal open class E2EIApiV7 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/E2EIApiV7.kt:24]:
```suggestion
    authenticatedNetworkClient: AuthenticatedNetworkClient
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/KeyPackageApiV7.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.network.api.v7.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.api.v6.authenticated.KeyPackageApiV6
+
+internal open class KeyPackageApiV7 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/KeyPackageApiV7.kt:25]:
```suggestion
    authenticatedNetworkClient: AuthenticatedNetworkClient
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/MLSMessageApiV7.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.network.api.v7.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.api.v6.authenticated.MLSMessageApiV6
+
+internal open class MLSMessageApiV7 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/MLSMessageApiV7.kt:25]:
```suggestion
    authenticatedNetworkClient: AuthenticatedNetworkClient
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/MLSPublicKeyApiV7.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.network.api.v7.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.api.v6.authenticated.MLSPublicKeyApiV6
+
+internal open class MLSPublicKeyApiV7 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/MLSPublicKeyApiV7.kt:25]:
```suggestion
    authenticatedNetworkClient: AuthenticatedNetworkClient
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/NotificationApiV7.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+package com.wire.kalium.network.api.v7.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.AuthenticatedWebSocketClient
+import com.wire.kalium.network.api.unbound.configuration.ServerConfigDTO
+import com.wire.kalium.network.api.v6.authenticated.NotificationApiV6
+
+internal open class NotificationApiV7 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient,
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v7/authenticated/NotificationApiV7.kt]:
```suggestion
    authenticatedNetworkClient: AuthenticatedNetworkClient,
```

File: network-model/src/commonMain/kotlin/com/wire/kalium/network/api/authenticated/client/ClientDTO.kt [hunk 6]
```diff
@@ -30,11 +36,28 @@ data class ClientDTO(
...
+    JsonTransformingSerializer<List<ClientCapabilityDTO>>(ListSerializer(ClientCapabilityDTO.serializer())) {
+    override fun transformDeserialize(element: JsonElement): JsonElement {
+        return when {
+            element is JsonObject && element.containsKey("capabilities") -> element["capabilities"]!!
+            element is JsonArray -> element
+            else -> throw SerializationException("Unexpected JSON format for capabilities")
+        }
+    }
+}
+
```
Comment (MohamadJaara) [inline network-model/src/commonMain/kotlin/com/wire/kalium/network/api/authenticated/client/ClientDTO.kt:60]: Can you please add a couple of tests for this serializer

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
suggestion to remove not used modifiers from generated api classes and missing test for the `CapabilitiesDeserializer`

**Review by MohamadJaara — APPROVED** [review]

**Review by ohassine — APPROVED** [review]

**Pull request merged**